tests/apm/mod.rs

```diff
@@ -2,10 +2,35 @@ use std::fs::{self, File};
 use std::io::{BufRead, BufReader};
 
 use bson::Bson;
-use mongodb::{Client, ClientOptions, CommandResult, ThreadedClient};
 use mongodb::db::ThreadedDatabase;
+use mongodb::{Client, ClientOptions, CommandResult, ThreadedClient};
 use rand;
 
+macro_rules! assert_eq_ignoring_ip {
+    ($expected: expr, $actual: ident) => {{
+        if $actual.contains("127.0.0.1") {
+            assert_eq!($expected, &$actual);
+        } else if $actual.contains("[::1]") {
+            let expected_string = &$expected.replace("127.0.0.1", "[::1]");
+            assert_eq!(expected_string, &$actual);
+        } else {
+            unimplemented!("Unable to understand which interface is binded");
+        }
+    }};
+}
```

Comment:
This breaks error messaging if actual returns something that doesn't include the IP address.  It also checks for a string in actual, and then modifies expected, which could have odd results.  I'd suggest logic more like 

```rust
let actual_string = if $actual.contains("[::1]") {
    $actual.replace("[::1]", "127.0.0.1")
} else {
    $actual
};
assert_eq!($expected, actual_string);
```